server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1936,7 +1937,12 @@ public VMTemplateVO createPrivateTemplateRecord(CreateTemplateCmd cmd, Account t
                     UserVmVO userVm = _userVmDao.findById(vmId);
                     if (userVm != null) {
                         _userVmDao.loadDetails(userVm);
-                        details.putAll(userVm.getDetails());
+                        Map<String, String> vmDetails = userVm.getDetails();
+                        vmDetails = vmDetails.entrySet()
+                                .stream()
+                                .filter(map -> map.getValue() != null)
+                                .collect(Collectors.toMap(map -> map.getKey(), map -> map.getValue()));
+                        details.putAll(vmDetails);
```

Comment:
@shwstppr changes LGTM ,to allow template creation from volume.

Do you know any reason to keep some keys (eg. _kvm.vnc.address_) with value null? Ideally, value in the details table shouldn't accept null.

Reply:
exactly, see my comment above.